apps/heft/src/plugins/CopyStaticAssetsPlugin.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { LegacyAdapters, FileSystem, Terminal } from '@rushstack/node-core-library';
+import * as glob from 'glob';
+import * as globEscape from 'glob-escape';
+import * as path from 'path';
+import * as chokidar from 'chokidar';
+
+import { Async } from '../utilities/Async';
+import { performance } from 'perf_hooks';
+import { IPluginPackage } from '../pluginFramework/IPluginPackage';
+import { Build, HeftCompilation } from '../pluginFramework/HeftCompilation';
+import { HeftConfiguration } from '../configuration/HeftConfiguration';
+import { ICompilePhase, ICopyStaticAssetsConfiguration } from '../cli/actions/BuildAction';
+import { PrefixProxyTerminalProvider } from '../utilities/PrefixProxyTerminalProvider';
+
+const PLUGIN_NAME: string = 'CopyStaticAssetsPlugin';
+
+export const copyStaticAssetsPlugin: IPluginPackage = {
```

Comment:
This happens to be a stateless plugin. But in general plugins will need to maintain state. 

Maybe we should always use a `class` or some slightly more structured pattern, so that if/when the person needs to add state, there is a natural place to track it without having to redesign your plugin?